File: OOAD_HW1/Q4/wolf.java
package Q4;

public class wolf extends canine {

	// Wolf's Name
	private String name;
	
	// Default Constructor, don't use.
	public wolf() {
		super("None");
		this.name = "None";
	}
	
	// Constructor for setting the wolf's name, and it's super-class' name.
	public wolf(String name) {
		super(name);
		this.name = name;
	}
}

File: OOAD_HW2/Part B/Wolf.cpp
#include <iostream>
#include <string>
#include <cstdlib>
#include <iomanip>

using namespace std;

// Wolf Class
class Wolf: public Canine
{
    protected:
        string name;
        string type;
        //NoiseBehavior *noise;

    public:
        Wolf(string aName, string aType);
};

// Behavior handled by interface via super class.
Wolf::Wolf(string aName, string aType): Canine(aName, aType)
{
    name = aName;
    type = aType;
}


